cylc/flow/task_job_mgr.py

```diff
@@ -1007,44 +1022,78 @@ def _set_retry_timers(
             except KeyError:
                 itask.try_timers[key] = TaskActionTimer(delays=delays)
 
-    def _simulation_submit_task_jobs(self, itasks, workflow):
-        """Simulation mode task jobs submission."""
+    def submit_nonlive_task_jobs(
+        self: 'TaskJobManager',
+        workflow: str,
+        itasks: 'List[TaskProxy]',
+        workflow_run_mode: str,
+    ) -> 'Tuple[List[TaskProxy], List[TaskProxy]]':
+        """Identify task mode and carry out alternative submission
+        paths if required:
+
+        * Simulation: Job submission.
+        * Skip: Entire job lifecycle happens here!
+        * Dummy: Pre-submission preparation (removing task scripts content)
+          before returning to live pathway.
+        * Live: return to main submission pathway without doing anything.
+
+        Returns:
+            lively_tasks:
+                A list of tasks which require subsequent
+                processing **as if** they were live mode tasks.
+                (This includes live and dummy mode tasks)
+            nonlive_tasks:
+                A list of tasks which require no further processing
+                because their apparent execution is done entirely inside
+                the scheduler. (This includes skip and simulation mode tasks).
+        """
+        lively_tasks: 'List[TaskProxy]' = []
+        nonlive_tasks: 'List[TaskProxy]' = []
         now = time()
-        now_str = get_time_string_from_unix_time(now)
+        now = (now, get_time_string_from_unix_time(now))
+
         for itask in itasks:
-            # Handle broadcasts
+            # Get task config with broadcasts applied:
             rtconfig = self.task_events_mgr.broadcast_mgr.get_updated_rtconfig(
                 itask)
+            # Apply task run mode
+            if workflow_run_mode in WORKFLOW_ONLY_MODES:
+                # Task run mode cannot override workflow run-mode sim or dummy:
+                run_mode = workflow_run_mode
+            else:
+                # If workflow mode is skip or live and task mode is set,
+                # override workflow mode, else use workflow mode.
+                run_mode = rtconfig.get('run mode', None) or workflow_run_mode
+
+            # Store the run mode of the this submission:
+            itask.run_mode = run_mode
+
+            # Submit nonlive tasks, or add live-like (live or dummy)
+            # tasks to list of tasks to put through live
+            # submission pipeline - We decide based on the output
+            # of the submit method:
+            submit_func = RunMode(run_mode).get_submit_method()
+            if not submit_func:
+                # Return to nonlive.
+                nonlive_mode = False
+            else:
+                nonlive_mode = submit_func(
+                    self, itask, rtconfig, workflow, now)
+
+            if nonlive_mode:
+                self.workflow_db_mgr.put_insert_task_states(
```

Comment:
@MetRonnie  - the submit number is incremented in the skip/sim job submit scripts, but this bit of code ensures the database is updated.